docs/contributing.md

```diff
@@ -1,70 +1,381 @@
+3. Follow the steps in the issue description to complete the issue.
+4. Make sure to comment your ETA and Availability when you first assign yourself.
+
+### **2.2 Working on an issue**
+
+#### **2.2.a Working on an issue(1): Verifying current branch**
+Once you have selected an issue to work on, create a branch for that issue. 
+
+Verify you are on the `main` branch. 
+
+```bash
+git branch
+```
+
+You will see a list of all of your branches. There will be a star (`*`) next to the branch that you are currently in. By default you should start on the `main` branch.
+
+If you are not currently in the `main` branch, run the following command to return to it:
+
+```bash
+git checkout main
+```
+
+Create a new branch where you will work on the issue. The branch name should include the issue number. For example, to create a new branch for issue 15 and change into it:
+
+```bash
+git checkout -b <new-branch-name>-15
+```
+
+#### **2.2.b Working on an issue(2): Prepare your changes to push to your repository**
+
+Once you are done with the work on your issue you will push it to your repository.  Before you can push your work to your repository, you will stage and commit your changes.  These two commands are similar to the save command used in other programs. 
+
+**Note:** If you are using Visual studios code you can use the Git graphical user interface to stage your changes. For instructions check out the [Git Gui Wiki](https://github.com/hackforla/website/wiki/Using-Git-GUI-(Graphical-user-Interface)-in-Visual-Studios-Code).
+
+**Note:** Remember to run the `precommit-checks.sh` script before each commit until it can be integrated.
+
+##### **i. Prepare repo changes (1): Use the `git add` command to stage your changes.** 
+
+**Make sure you are on your issue branch (instead of `main`)**
+
+This command prepares your changes before you commit them. You can stage files one at a time using the filename. 
+
+Run this command if you want to **add changes from a specific file to your commit record**: 
+```bash
+git add “filename.ext”
+```
+
+Run this command if you want to **add all changes to all file(s) to your commit record**: 
+```bash
+git add .
+```
+
+##### **ii. Prepare repo changes (2): Use the `git status` command to see what files are staged.**
+
+This command will list the files that have been staged with green text. These are the files that will be committed (saved) when you run the next command, `git commit`. Please be sure all your staged changes are relevant to the issue you are working on. If you accidentally included unrelated changes, please unstage them before making this commit, and then make a new commit for the unrelated changes. (The commands for unstaging commits are provided in the output of your `git status` command.)
+      
+```bash
+git status
+```
 
-### Testing
+##### **iii. Prepare repo changes (3): Use the `git reset HEAD` command to remove a staged file.**
```

Comment:
@ethanstrominger ~ i think the command should be `git checkout <file-name>`. This is an example of something that should request changes (as well as leaving a comment), because it's incorrect as is